simplefi-balancer-v2/schema.graphql

```diff
@@ -0,0 +1,232 @@
+
+# It is required because we want to keep track of closed positions as well
+# We can not use same ID for a user's new position after existing one has been closed
+# To be able to fetch current open position for a user + market + position type we need
+# ID to be dependent on only these three things. There three things will be same for all positions
+# of a specific user in specfic market therefore we need this AccountPosition entity
+type AccountPosition @entity {
+  id: ID! # {userAddress}{marketAddress}{positionType}
+  positionCounter: BigInt!
+  positions: [Position!]! @derivedFrom(field: "accountPosition")
+}
+
+# We can not keep balances and roi updated all the time. 
+# We can update these only when the account of the position makes some changes in the position
+# at others time when market variables are changing because of other's actions then we won't update these
+type Position @entity {
+  id: ID! # {accountPositionId}{autoIncrement}
+  accountPosition: AccountPosition!
+  account: Account!
+  accountAddress: String!
+  market: Market!
+  marketAddress: String!
+  positionType: PositionType!
+  outputTokenBalance: BigInt!
+  inputTokenBalances: [String!]!    # TokenBalance string format
+  rewardTokenBalances: [String!]!   # TokenBalance string format
+  transferredTo: [String!]!
+  closed: Boolean!
+  blockNumber: BigInt!
+  timestamp: BigInt!
+  historyCounter: BigInt!
+  history: [PositionSnapshot!]! @derivedFrom(field: "position")
+}
+
+# Created before every update (transaction) to Position
+type PositionSnapshot @entity {
+  id: ID! # {positionId}{autoIncrement}
+  position: Position!
+  transaction: Transaction!
+  outputTokenBalance: BigInt!
+  inputTokenBalances: [String!]!    # TokenBalance string format
+  rewardTokenBalances: [String!]!   # TokenBalance string format
+  transferredTo: [String!]!         # TokenBalance string format
+}
+
+# END - Common entities for all protocols
+
+# START - Balancer V2 protocol entities
+
+enum PoolSpecialization {
+  GENERAL
+  MINIMAL_SWAP_INFO
+  TWO_TOKEN
+}
+
+enum PoolType {
+  WEIGHTED_POOL
+  WEIGHTED_POOL_2_TOKENS
+  STABLE_POOL
+  METASTABLE_POOL
+  LIQUIDITY_BOOTSTRAPPING_POOL
+  INVESTMENT_POOL
+}
+
+type Pool @entity {
+  id: ID! # {poolId}
+  address: String
+  factory: Account
+  tokens: [Token!]
+  reserves: [BigInt!]
+  totalSupply: BigInt
+  blockNumber: BigInt!
+  timestamp: BigInt!
+  poolSpecialization: PoolSpecialization
+  poolType: PoolType
+}
+
+type PoolId @entity {
+  id: ID! # {poolAddress}
+  poolId: String
+}
+
+type AccountLiquidity @entity {
+  id: ID! # {poolAddress}{accoutnAddress}
+  pool: Pool!
+  account: Account!
+  balance: BigInt!
+}
+
+type Mint @entity {
+  id: ID! # {poolAddress}{trasactionHash}
+  pool: Pool
+  to: Account
+  liquityAmount: BigInt
+  amounts: [BigInt!]
+  transferEventApplied: Boolean!
+  poolBalanceEventApplied: Boolean!
+}
+
+# @todo: Would be better to merge Mint and Burn and add a type?
```

Comment:
@NelsonGaldeman Can be done by appending type in ID but what if there is both Mint and Burn in same transaction?